feat(filters): dynamic filter components - part 2

Birkbjo opened the pull request.

Birkbjo commented on src/components/sectionList/filters/filterSelectors/PublicAccessFilter.tsx:
```diff
@@ -0,0 +1,19 @@
+import React from 'react'
+import { ConstantSelectionFilter } from './ConstantSelectionFilter'
+
+const constants = {
+    'rw------': 'Public can edit',
+    'r-------': 'Public can view',
+    '--------': 'Public cannot access',
+}
```
Note that this currently worldn't work for objects that also have "data sharing" (like category options). This was just a simple implementation for now.

Mohammer5 reviewed (commented):
I just gave it a try. I added the "public access" filter and removed the "domain type" filter. But the domain type filter is still showing.

Birkbjo commented:
> I just gave it a try. I added the "public access" filter and removed the "domain type" filter. But the domain type filter is still showing.

I just did this on the preview branch and it works. Will will update and fix conflicts in this branch though.

Did you select a domain type value? If a filter has a value, it will always show.

Mohammer5 commented:
Yeah, now it works for me as well. Not sure what happened, maybe I did select a value.
I suppose we could disable selected options in the transfer that have a value, but that's a question for @cooper-joe. For now it's fine I guess

Birkbjo commented on src/lib/dataStore/useDataStore.ts:
```diff
@@ -102,6 +102,7 @@ export function useDataStoreValuesQuery<
         queryKey: [query],
         queryFn: createBoundQueryFn(engine),
         placeholderData,
+        // keepPreviousData: true,
```
Todo: fix
